src/object.jl

```diff
@@ -0,0 +1,35 @@
+import CloudBase: AbstractStore, CloudCredentials
+
+struct Object{T <: AbstractStore, C <: Union{CloudCredentials, Nothing}}
+    store::T
+    credentials::C
```

Comment:
does credentials need to be a type param?

One (of many) annoying thing about `S3Path`s is that they're parametric on the "config" (i.e. credentials + region) https://github.com/JuliaCloud/AWSS3.jl/issues/240

Reply:
Yeah, that's probably better. I might just make it:
```julia
credentials::Union{Nothing, AWS.Credentials, Azure.Credentials}
```
since we know we'll have a very small # of concrete types to worry about here.